### PR TITLE
Use the multiwatcher worker in the apiserver and model cache.

### PR DESCRIPTION
**Pull request by howbazaar**
This requires updates to the manifolds. For the model cache it replaces
the need to get the state pool. For the apiserver, it needs the
multiwatcher worker in addition to its other pieces.

The multiwatcher worker is now started in the apiserver tests and
in the dummy provider to make the multiwatchers available.

## QA steps

Changes are all internal, no user facing changes. As long as the unit tests all continue to pass and we can deploy a bundle, we should be all good.

## Documentation changes

Everything is internal, no user facing change.

Except that now if you call Watch on a model, you also get a model update. But I don't think this is documented anywhere anyway.

### CODE REVIEW

**Comment by howbazaar** [comment]
!!build!!

File: worker/modelcache/manifold_test.go [hunk 1]
```diff
@@ -141,46 +137,12 @@ func (s *ManifoldSuite) TestNewWorkerArgs(c *gc.C) {
...
-}
-
-// Return an invalid but non-zero state pool pointer.
-// Is only ever used for comparison.
-func (f *fakeStateTracker) Use() (*state.StatePool, error) {
-	return f.pool(), nil
-}
-
-func (f *fakeStateTracker) pool() *state.StatePool {
-	return (*state.StatePool)(unsafe.Pointer(f))
```
Comment (hpidcock) [inline worker/modelcache/manifold_test.go]: ohmy

Comment (howbazaar) [reply worker/modelcache/manifold_test.go]: heh, yeah, I just wanted a non-zero pointer

**Comment by howbazaar** [comment]
$$merge$$

**Comment by hpidcock** [comment]
Kube core bundle deployed on lxd nicely.

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**